app/src/main/java/ch/epfl/sdp/Event.java

```diff
@@ -110,6 +149,29 @@ public String getAddress() {
         return mAddress;
     }
 
+    @Override
+    public String toString(){
+        return "Title:" + mTitle + "\nDescription:" + mDescription + "\nDate:" + mDate + "\nAddress:"
+                + mAddress + "\nLatitude:" + mLatitude + "\nLongitude:" + mLongitude
+                + "\nimageID:" + mImageId;
+    }
+
+    @Override
+    public boolean equals(Object o){
+        if(this == o)
+            return true;
+        if(o == null)
+            return false;
+
+        if(getClass() != o.getClass())
+            return false;
```

Comment:
replace to 
```
if (!o || getClass() != o.getClass())
   return false 
if (this == o)
   return true 
```

Reply:
pas moi mais je crois que c'est en raison de la complexité qui augmente trop pour code climate